Source/DafnyRuntime/DafnyRuntime.go

```diff
@@ -597,6 +597,8 @@ func (seq Seq) UniqueElements() Set {
 func (seq Seq) String() string {
   if seq.isString {
     s := ""
+    // Note this doesn't produce the right string in UTF-8,
+    // since it converts surrogates independently.
```

Comment:
Isn't that a bug?

Reply:
Definitely, just one I didn't want to bring in scope - I wasn't confident it was reasonable to depend on the necessary Go package to decode UTF-16. But I'll at least cut an issue (plus C++ is an even bigger mess since it uses the 8-bit `char` type).

Reply:
+1, thanks

Reply:
Yup. And unfortunately the more I test the more I find ugly inconsistencies on corner cases. :(

Reply:
https://github.com/dafny-lang/dafny/issues/2929